### ISSUE TITLE
Add Kaguya TC from ISIS Cube Driver

### ISSUE THREAD

**Issue by jlaura** [USGS-Astrogeology/ale#493]
According to #375 ALE can generate a valid ISD from the PDS3 label, but not from the ISIS label. There are instances when it is necessary to use the ISIS cube (for example after adjustment). Please add a Kaguya TC driver (for all 3 flavors of TC: monoscopic, stereoscopic, and SP support) that accepts the ISIS cube as the input.

**Comment by acpaquette** [comment]
@jlaura There is a PR for this in #487 it just needs a review

**Comment by jlaura** [comment]
Yup. ISISLabel, ISISSpice. I tried out this PR and it failed on my data because it is looking for metakernels. I understand that TC doesn't ship with metakernels (no complaint from me there).

**Comment by jessemapel** [comment]
There are some docs here: https://github.com/USGS-Astrogeology/ale/wiki/Adding-New-Drivers

You can probably start with the IsisLabel NaifSpice driver, swap the mix-in to from NaifSpice to IsisSpice, and then take out most of the properties that use spice. The ISIS attached SPICE format is much more standardized than the kernel data, so the mix-in can handle most things.

Testing is tricky, but @acpaquette may have some docs he can point you to.

**Comment by jlaura** [comment]
I started to implement this and it looks really straight forward as you indicated @jessemapel. It looks like I am going to need to furnish the IK kernels in order to get the distortion coefficients. What is the preferred method to do this? I see that data_naif has a kernels property that will attempt to furnish, but that is hard coded to check for ALESPICEROOT or use the `prop` kwarg. 

The solution that seems to be working also feels really hacky. I added a kernels property into the concrete driver a la:

```python
    @property
    def kernels(self):
      if not hasattr(self, '_kernels'):
        self._kernels = util.get_kernels_from_isis_pvl(self.label['IsisCube'])
        print(self._kernels)
        [spice.furnsh(k) for k in self._kernels]
```

and then call `self.kernels` inside the properties that might need the kernels furnished (since the call order it not guaranteed).


PR in #501. Happy to discuss the implementation over there!
